### PR TITLE
Fix #1247: Remapping a property to a field results in invalid parameter capitalization

### LINKED ISSUE #1247: Remapping a property to a field results in invalid parameter capitalization
``` C#
    [CodeGenModel("Response")]
    public partial class MetricQueryResult
    {
        [CodeGenMember("Timespan")]
        private readonly string _timespan;
```

Results in the following constructor:

```C#
internal MetricQueryResult(string Timespan, IEnumerable<Metric> metrics)
        {
```

### PR DESCRIPTION
**Pull request by AlexanderSher**

### CODE REVIEW

File: test/AutoRest.TestServer.Tests/TypeSchemaMappingTest.cs [hunk 1]
```diff
@@ -100,7 +100,7 @@ public void ObjectsAreMappedToSchemas()
         [Test]
         public void NullableParamsAsNull()
         {
-            var model = new RenamedModelStruct("test", null, null);
```
Comment (pakrym) [inline test/AutoRest.TestServer.Tests/TypeSchemaMappingTest.cs:103]: should we use an explicit parameter name here to test that we get the expected name?

Comment (pakrym) [reply test/AutoRest.TestServer.Tests/TypeSchemaMappingTest.cs:103]: Maybe worth adding something to https://github.com/Azure/autorest.csharp/blob/feature/v3/test/AutoRest.TestServer.Tests/Common/StringExtensionsTests.cs

**Comment by pakrym** [comment]
This PR seems to have some unrelated changes.

**Review by pakrym — APPROVED** [review]

File: test/TestServerProjects/lro/Generated/LROsRestClient.cs [hunk 2]
```diff
@@ -86,60 +86,6 @@ public Response Put200Succeeded(Product product = null, CancellationToken cancel
...
-            await _pipeline.SendAsync(message, cancellationToken).ConfigureAwait(false);
-            var headers = new LROsPatch200SucceededIgnoreHeadersHeaders(message.Response);
-            switch (message.Response.Status)
-            {
-                case 200:
-                    return ResponseWithHeaders.FromValue(headers, message.Response);
-                default:
-                    throw await _clientDiagnostics.CreateRequestFailedExceptionAsync(message.Response).ConfigureAwait(false);
-            }
-        }
```
Comment (ShivangiReja) [inline test/TestServerProjects/lro/Generated/LROsRestClient.cs:124]: autorest.testserver version has been updated. Please ensure to run `npm install` first before you regenerate all the test projects.

**Pull request merged**